modules/sqs/src/main/scala/com/rewardsnetwork/pureaws/sqs/SimpleSqsClient.scala

```diff
@@ -41,19 +41,27 @@ trait SimpleSqsClient[F[_]] {
   /** Delete a message from AWS SQS. */
   def deleteMessage(rawReceiptHandle: String, queueUrl: String): F[Unit]
 
-  /** Send a message to an SQS queue. Message delay is determined by the queue settings.
+  /** Send a message with attributes to an SQS queue. Message delay is determined by the queue settings.
     * @return
     *   The message ID string of the sent message.
     */
-  def sendMessage(queueUrl: String, messageBody: String): F[String]
+  def sendMessage(
+      queueUrl: String,
+      messageBody: String,
+      messageAttributes: Map[String, MessageAttributeValue]
+  ): F[String]
 
-  /** Sends a message to an SQS queue. Allows specifying the seconds to delay the message (valid values between 0 and
-    * 900).
+  /** Send a message with attributes to an SQS queue. Allows specifying the seconds to delay the message (valid values
+    * between 0 and 900).
     * @return
     *   The message ID string of the sent message.
     */
-  def sendMessage(queueUrl: String, messageBody: String, delaySeconds: Int): F[String]
-
+  def sendMessage(
+      queueUrl: String,
+      messageBody: String,
+      delaySeconds: Int,
+      messageAttributes: Map[String, MessageAttributeValue]
+  ): F[String]
```

Comment:
I see what you're saying about the problem with default arguments. What if instead, you added two more overrides without the `messageAttributes`, so now there will be four. One for just the URL and body, one with delay seconds, and two mirroring those but also with `messageAttributes` as well with no default. I believe that might work.